pkg/clusteragent/clusterchecks/dispatcher_main.go

```diff
@@ -61,6 +73,23 @@ func (d *dispatcher) remove(config integration.Config) {
 		return // Ignore non cluster-check configs
 	}
 	digest := config.Digest()
-	log.Debugf("removing configuration %s:%s", config.Name, digest)
+	log.Debugf("Removing configuration %s:%s", config.Name, digest)
 	d.removeConfig(digest)
 }
+
+// cleanupLoop is the cleanup goroutine for the dispatcher.
+// It has to be called in a goroutine with a cancellable context.
+func (d *dispatcher) cleanupLoop(ctx context.Context) {
+	expireTicker := time.NewTicker(time.Duration(d.nodeExpirationSeconds/2) * time.Second)
+	defer expireTicker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-expireTicker.C:
+			expired := d.expireNodes()
```

Comment:
I would rename expired to orphanedConfigs (so that it's not ambiguous with the expireNode method)